devel/libecl/src/ecl_grid.c

```diff
@@ -5717,11 +5721,24 @@ static void ecl_grid_fwrite_EGRID__( ecl_grid_type * grid , fortio_type * fortio
   {
     {
       ecl_grid_assert_coord_kw( grid );
-      ecl_kw_fwrite( grid->coord_kw , fortio );
+      if(is_metric != grid->is_metric){
+          ecl_kw_type * coord_kw = ecl_kw_alloc_data_copy(grid->coord_kw);
+          ecl_kw_scale_float(coord_kw, METER_TO_FEET_SCALE_FACTOR);
```

Comment:
Since we are in this block we *know* that `output_metric != grid->is_metric` - i.e. a conversion should take place. However we must check one of the variables to know which way the conversion should go:

```
double scale_factor = if (grid->is_metric) ? METER_TO_FEET_SCALE_FACTOR : 1 / METER_TO_FEET_SCALE_FACTOR;
```